core/parameters/src/config_store.rs

```diff
@@ -432,4 +435,27 @@ mod tests {
         let config = store.get_config(PROTOCOL_VERSION);
         assert_eq!(config.witness_config.main_storage_proof_size_soft_limit, u64::MAX);
     }
+
+    // This test ensures that the gas and compute costs for touching and reading
+    // a trie node remain equal. Please do not remove this check as the code
+    // may rely on this behavior after the version upgrade. For example, the
+    // code may no longer differentiate between touching a trie node for the
+    // first time and reading it again.
+    #[test]
+    fn test_equalize_trie_node_touch_and_read_cost() {
```

Comment:
Not sure if this is the best location for this test?

Reply:
contract runtime (possibly ext.rs) code that relies on these being the same would be more appropriate.